tests/cases/example.result

```diff
@@ -0,0 +1,18 @@
+CREATE TABLE demo (name string TAG, value double NOT NULL, t timestamp NOT NULL, TIMESTAMP KEY(t)) ENGINE=Analytic with (enable_ttl='false');
```

Comment:
It seems a little inconvenient to repeat the input SQL.

Maybe we can only use one file(same with this result format), and let this framework to extract SQL out automatically. Any thoughts?

Reply:
```test
# Simple sum.
eval instant at 50m SUM BY (group) (http_requests{job="api-server"})
  {group="canary"} 700
  {group="production"} 300

eval instant at 50m SUM BY (group) (((http_requests{job="api-server"})))
  {group="canary"} 700
  {group="production"} 300

# Test alternative "by"-clause order.
eval instant at 50m sum by (group) (http_requests{job="api-server"})
  {group="canary"} 700
  {group="production"} 300
```
This is what PromQL test cases look like.
- https://github.com/ceresdb/prometheus/blob/16bcbdaac510f1e5be7e532f7060a8a6f95e05d5/promql/testdata/aggregators.test#L11

Reply:
The ".result" file is not intended to be written out. Instead it's generated (usually it's renamed from a ".out" file). Thus it's ok to be redundant here. Repeat SQL can improve the readability of this file.

Reply:
Using one file have the same effect with what you said, and without any redundancies.

```org
select * from xx;

--result begin---
xxx
--result end---

select * from yy;

--result begin---
yyy
--result end---
```


When write a new test case, users can just write SQL, leaving the result part empty, then run the tests to get the results(in `out` file), then copy & paste that to `result` part to get job done.

Reply:
This is feasible in implementation. But the query result can be a lot. Consider an example: [SQL file](https://github.com/MariaDB/server/blob/10.10/mysql-test/main/intersect.test) and [result file](https://github.com/MariaDB/server/blob/10.10/mysql-test/main/intersect.result). I think it is hard to even find out a SQL you want in the result file. And the input SQLs in one file are usually causally related, putting results between them may make it harder to read.

Reply:
> And the input SQLs in one file are usually causally related, putting results between them may make it harder to read

This make sense to me. Let's wait see what others opinion is.